Fix table height calculation for captions with orthogonal writing modes

Ahmad-S792 opened the pull request:
#### 5ef3ef6b9fa8d953be31c1ef0e3f91edd5b1c63e
<pre>
Fix table height calculation for captions with orthogonal writing modes
<a href="https://bugs.webkit.org/show_bug.cgi?id=304714">https://bugs.webkit.org/show_bug.cgi?id=304714</a>
<a href="https://rdar.apple.com/167220730">rdar://167220730</a>

Reviewed by Alan Baradlay.

When a caption has a perpendicular writing mode to its table, use the
caption&apos;s logical width (in caption&apos;s writing mode) as the contribution
to the table&apos;s logical height (in table&apos;s writing mode).

* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::layoutCaption):
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/tentative/caption-expected.txt: Progression

Canonical link: <a href="https://commits.webkit.org/305110@main">https://commits.webkit.org/305110@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9909aa7c4d326da55d56b2ad99df2487872ebb3a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137548 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9909 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/145305 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/90521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6ae6b333-62a7-4577-9055-625ffdbfec8a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/139420 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10612 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/10039 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105193 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/90521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e72263c8-b379-4b88-a2a2-97688978e30e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140493 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7875 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123273 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/86046 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/042fc9d4-d579-4092-a638-c2188883e540) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/7511 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5233 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5888 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116881 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41429 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/148069 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9591 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41983 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/113573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9609 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/8085 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/113917 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28907 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7430 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119512 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/64247 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9640 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37565 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9371 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/73205 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9580 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9432 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderTable.cpp:
```diff
@@ -422,7 +422,15 @@ void RenderTable::layoutCaption(RenderTableCaption& caption)
+    // When caption has a different writing mode, we need to use the caption's size in the table's writing mode.
+    LayoutUnit captionLogicalHeightInTableWritingMode;
+    if (caption.writingMode().isOrthogonal(writingMode())) {
+        // Caption is perpendicular to table - use caption's logical width as table's logical height contribution.
```
I am not sure if this comment is particularly helpful here.

Ahmad-S792 replied:
@alanbaradlay - thanks - I updated and removed this comment, you are right, commit message is almost same, so it might be duplicate here.

Ahmad-S792 committed:
Fix table height calculation for captions with orthogonal writing modes
https://bugs.webkit.org/show_bug.cgi?id=304714
rdar://167220730

Reviewed by Alan Baradlay.

When a caption has a perpendicular writing mode to its table, use the
caption's logical width (in caption's writing mode) as the contribution
to the table's logical height (in table's writing mode).

* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::layoutCaption):
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/tentative/caption-expected.txt: Progression

Canonical link: https://commits.webkit.org/305110@main
